apps/artemis-web/components/hijack-table/hijack-table.tsx

```diff
@@ -578,17 +580,27 @@ const HijackTableComponent = (props) => {
     const currentIndex = page * sizePerPage;
     setPage(page);
     setSizePerPage(sizePerPage);
-    if (currentIndex) setOffsetState(currentIndex);
-    if (sizePerPage) setLimitState(sizePerPage);
+
+    setOffsetState(currentIndex);
+    setLimitState(sizePerPage);
     if (sortOrder) {
       setSortColumnState(sortField);
       setSortState(sortOrder);
     }
     if (filters) {
-      const key = Object.keys(filters)[0];
-      if (filters[key])
-        setStateValues({ ...stateValues, [key]: filters[key].filterVal });
-      else setStateValues({ ...stateValues, [key]: '' });
+      const keys = Object.keys(filters);
+
+      keys.forEach((key) => {
+        if (filters[key])
+          setStateValues({ ...stateValues, [key]: filters[key].filterVal });
+        else setStateValues({ ...stateValues, [key]: '' });
+      });
+
+      if (currentIndex && !compareObjects(filters, columnFilter)) {
+        setPage(0);
+        setOffsetState(0);
```

Comment:
can we reuse some the same code used in the tables over and over (update + hijack tables)?

Reply:
yes we can

Reply:
OK, please mark in the PR what can be reused e.g., as a function so that we reduce the probability of the same bug in two or more different places. This is sth general to consider for the code, but for now try to apply it here. It is also OK if it is not easily isolated in a function to leave as-is. In the latter case, you can resolve the comment.